MARIE_SEQ2SEQ/training/README.md

```diff
@@ -0,0 +1,110 @@
+1. Install the remaining packages required by the project `pip install -r requirements.txt`.
+1. For access to any private models or datasets on HuggingFace (e.g. Llama2), ensure that you have the necessary read privileges. Generate an [access token](https://huggingface.co/docs/hub/security-tokens) and make this an environment variable: `export HF_ACCESS_TOKEN=<huggingface-access-token>`.
+
+## Data format
+
+```{json}
+[
+   {
+      "question": str,
+      "sparql_query": str,
+      "sparql_query_compact: str
+   }
+]
+```
+
+## Fine-tuning
+
+### Weight & Biases configuration for monitoring
+
+1. [Sign up](https://wandb.ai/site) for a Weights & Biases account. 
+1. Locate your API key on the [Authorize page](https://wandb.ai/authorize).
+1. Execute the following command and specify the API key.
+   ```
+   wandb login
+   ```
+1. Set the Weights & Biases project name as an environment variable.
+   ```
+   export WANDB_PROJECT=marie
+   ```
+
+### Training
+
+For access to parameter-efficient fine-tuning methods such as LoRA, `pip install bitsandbytes==0.40.2 peft==0.4.0`.
+
+See [`finetune.sh`](./scripts/finetune.sh) for example usage. The script will save the weights of the fine-tuned model in the specified `output_dir`.
+
+For a list all arguments that can be passed to the fine-tune script, see classes `arguments_schema.ModelArguments`, `arguments_schema.DatasetArguments` and `transformers.TrainingArguments`.
+
+## Conversion of Hugging Face models to other formats
+
+```
+python convert_model.py <path-to-hf-model-directory> <path-to-output-directory> --target_format [onmt|ov|ort]
+```
+
+## Inference
+
+Additional dependencies:
+- To run inference on different runtimes:
+  - OpenNMT: `pip install ctranslate2 pyonmttok`
+  - OpenVINO: `pip install optimum==1.12.0 && pip install  --upgrade-strategy eager optimum[openvino,nncf]`
+  - ONNX Runtime for CPU: `pip install optimum==1.12.0 && pip install --upgrade-strategy eager install optimum[onnxruntime]`
+  - ONNX Runtime for GPU: `pip install optimum==1.12.0 && pip install optimum[onnxruntime-gpu]`
+    - TensorRT: 
+    - [CUDA toolkit](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html)
+    - [cuDNN](https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html)
+    - [TensorRT](https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html) (installation from tar ball recommended)
+- To enable memory profiling with the command line argument `--do_profile`, run `pip install memory-profiler==0.61.0`.
+
+See [`inference.sh`](./scripts/inference.sh) for example usage. The script will generate a json file containing  
+
+For a list of all arguments that can be passed to the inference script, see classes `arguments_schema.ModelArguments`, `arguments_schema.DatasetArguments`, and `arguments_schema.InferenceArguments`.
+
+## Evaluation
+
+```
+python evaluate.py predictions.json evaluation_results.json --do_correct_spans --do_correct_relations
+```
```

Comment:
A sentence or two on what is being evaluated and how would be highly appreciated.

Reply:
I've added an elaboration on what evaluation metrics are computed.